src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -2115,6 +2115,13 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                     type = null;
                     nullability = new NullabilityInfo(CodeAnalysis.NullableAnnotation.NotAnnotated, CodeAnalysis.NullableFlowState.NotNull);
                 }
+                else if (highestBoundExpr is BoundConvertedSwitchExpression e)
+                {
+                    type = e.NaturalTypeOpt;
+                    convertedType = e.Type;
+                    convertedNullability = e.TopLevelNullability;
+                    conversion = e.Conversion.IsValid ? e.Conversion : Conversion.NoConversion;
```

Comment:
>e.Conversion.IsValid ? e.Conversion : Conversion.NoConversion [](start = 33, length = 61)

Can we simply use `e.Conversion` always?

Reply:
No. The compiler records an invalid target-typed conversion when one of the arms doesn't convert. But for the purposes of the API, we just say there is no conversion.

---
In reply to: [438547559](https://github.com/dotnet/roslyn/pull/45069#discussion_r438547559) [](ancestors = 438547559)